11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+			shaders[1] = createShader("app_resources/kernel_fft_second_axis.hlsl", workgroupSize, elementsPerThread, bloomScale);
+			shaders[2] = createShader("app_resources/kernel_spectrum_normalize.hlsl", workgroupSize, elementsPerThread, bloomScale);
+
+			// -------------------------------------------
+
+			// Create compute pipelines - First axis FFT -> Second axis FFT -> Normalization
+			IGPUComputePipeline::SCreationParams params[3];
+			// First axis FFT
+			params[0].layout = kernelFirstAxisFFTPipelineLayout.get();	
+			// Second axis FFT -  since no descriptor sets are used, just keep the same pipeline layout to avoid having yet another pipeline layout creation step
+			params[1].layout = kernelFirstAxisFFTPipelineLayout.get();
+			// Normalization
+			params[2].layout = kernelNormalizationPipelineLayout.get();
+			// Common
+			for (auto i = 0u; i < 3; i++) {
+				params[i].shader.entryPoint = "main";
+				params[i].shader.shader = shaders[i].get();
+				params[i].shader.requireFullSubgroups = true;
+			}
+			
+			smart_refctd_ptr<IGPUComputePipeline> pipelines[3];
+			if(!m_device->createComputePipelines(nullptr, { params, 3 }, pipelines))
+				return logFail("Failed to create Compute Pipelines!\n");
+
+			// Push Constants - only need to specify BDAs here
+			PushConstantData pushConstants;
+			pushConstants.colMajorBufferAddress = m_colMajorBufferAddress;
+			pushConstants.rowMajorBufferAddress = m_rowMajorBufferAddress;
+
+			m_computeCmdBuf->begin(IGPUCommandBuffer::USAGE::ONE_TIME_SUBMIT_BIT);
+
+			// First Axis FFT
+			m_computeCmdBuf->bindComputePipeline(pipelines[0].get());
+			m_computeCmdBuf->bindDescriptorSets(asset::EPBP_COMPUTE, pipelines[0]->getLayout(), 0, 1, &kernelFFTDescriptorSets[0].get());
+			m_computeCmdBuf->pushConstants(pipelines[0]->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, 0u, sizeof(pushConstants), &pushConstants);
+			// One workgroup per 2 columns
+			m_computeCmdBuf->dispatch(paddedKerDim.width / 2, 1, 1);
+
+			// Pipeline barrier: wait for first axis FFT before second axis can begin
+			IGPUCommandBuffer::SPipelineBarrierDependencyInfo pipelineBarrierInfo = {};
+			decltype(pipelineBarrierInfo)::buffer_barrier_t bufBarrier = {};
+			pipelineBarrierInfo.bufBarriers = { &bufBarrier, 1u };
+			
+			// First axis FFT writes to colMajorBuffer
+			bufBarrier.range.buffer = m_colMajorBuffer;
+
+			// Wait for first compute write (first axis FFT) before next compute read (second axis FFT)
+			bufBarrier.barrier.dep.srcStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT;
+			bufBarrier.barrier.dep.srcAccessMask = ACCESS_FLAGS::MEMORY_WRITE_BITS;
+			bufBarrier.barrier.dep.dstStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT;
+			bufBarrier.barrier.dep.dstAccessMask = ACCESS_FLAGS::MEMORY_READ_BITS;
+
+			m_computeCmdBuf->pipelineBarrier(asset::E_DEPENDENCY_FLAGS(0), pipelineBarrierInfo);
+
+			// Now do second axis FFT - no need to push the same constants again I think
```

Comment:
indeed no need, but I want you to use 1 descriptor set to rule all, so you may need to push a somewhat modified constant